scripts/actors-models-graph.py

```diff
@@ -0,0 +1,185 @@
+                    produces=tuple(p.serialize()['name'] for p in actor.produces),
+                    tags=filter(lambda elem: elem != 'IPUWorkflowTag', tuple(t.serialize()['name'] for t in actor.tags))
+                    #tags=filter(lambda elem: elem != 'ipu', tuple(t.name for t in actor.tags))
+                  ) for actor in repository.actors]
+
+    # mapping from tag to actors
+    # key: tag, value: actor name
+    tags = {}
+    for actor in actors:
+        for tag in actor.tags:
+            if tag not in tags:
+                tags[tag] = set()
+            tags[tag].add(actor.name)
+
+    # models that are consumed/produced by some actor
+    models_consumed = set()
+    models_produced = set()
+    for actor in actors:
+        models_consumed.update(actor.consumes)
+        models_produced.update(actor.produces)
+
+    # models which are produced/consumed but no actor consumes/produces them
+    models_missing_consumer = models_produced.difference(models_consumed)
+    models_missing_producer = models_consumed.difference(models_produced)
+
+    # create dummy actors that consumes/produces models which lacks proper consumer/producer
+    missing_consumer_actors = [Actor(name=model + 'Consumer',
+                                    consumes=(model,),
+                                    produces=(),
+                                    tags=()
+                                    ) for model in models_missing_consumer]
+    missing_producer_actors = [Actor(name=model + 'Producer',
+                                    consumes=(),
+                                    produces=(model,),
+                                    tags=()
+                                    ) for model in models_missing_producer]
+
+    actors.extend(missing_consumer_actors)
+    actors.extend(missing_producer_actors)
+
+    # mapping from consumed models to actors
+    # key: consumed model, value: set of actor names that consumes the key
+    model_consumers = {}
+    for actor in actors:
+        for model in actor.consumes:
+            if model not in model_consumers:
+                model_consumers[model] = set()
+            model_consumers[model].add(actor.name)
+
+    with open(args.dot, 'w') as file_handle:
+        # opening tag
+        file_handle.write('digraph g {\n')
+
+        # basic directed graphs - only actors (nodes) and models (edges)
+        basic_graph(actors, model_consumers, missing_producer_actors, missing_consumer_actors, file_handle)
+
+        # set graph parameters
+        node_distance(file_handle)
+
+        # uncomment following line for clustering of actors by tags
+        # warning: the edges are too close to each other, so it's usually hard to say
+        # which label belongs to which edge
+        if args.tags:
+            clusters(tags, file_handle)
+
+        # closing tag
+        file_handle.write('}\n')
+
+    print >> sys.stderr, "To generate figure with graph, use dot tool:"
```

Comment:
This is not Python3 compatible, this should be fixed however
```suggestion
    sys.stderr.write("To generate figure with graph, use dot tool:\n")
```